sinks/signalfx/signalfx.go

```diff
@@ -17,34 +18,98 @@ import (
 	"github.com/stripe/veneur/trace"
 )
 
-type SignalFxSink struct {
-	client           dpsink.Sink
-	endpoint         string
-	hostnameTag      string
-	hostname         string
-	commonDimensions map[string]string
-	log              *logrus.Logger
-	traceClient      *trace.Client
-	excludedTags     map[string]struct{}
+// collection is a structure that aggregates signalfx data points
```

Comment:
It might be good to use SFx's [AsyncMultiTokenSink](https://github.com/signalfx/golib/blob/4bab910cc50c58be2c0a848217a137bb5bf3090e/sfxclient/multitokensink.go#L449) for this? Then they do the heavy lifting for you. You'd still need to configure it and figure out what token to use, then call [AddDatapointsWithToken](https://github.com/signalfx/golib/blob/4bab910cc50c58be2c0a848217a137bb5bf3090e/sfxclient/multitokensink.go#L502) to add it.  It's also async, which creates some new fun. :)

Reply:
That's really cool! (I probably should read the api docs of libs I use, ahahah!), but I think it won't come out quite as nicely as this: Currently we can quite easily test the routing functionality, but with the AsyncMultiTokenSink, we'd have to parse http requests it makes or something. Hrm.